docs/develop/python/user-metadata.mdx

```diff
@@ -0,0 +1,259 @@
+
+## Adding Metadata
+
+### Workflow
+
+#### Starting a Workflow
+
+When starting a workflow, you can provide a static summary and details to help identify the workflow in the UI:
+
+```python
+# Start a workflow with static summary and details
+handle = await client.start_workflow(
+    YourWorkflow.run,
+    "workflow input",
+    id="your-workflow-id",
+    task_queue="your-task-queue",
+    static_summary="Order processing for customer #12345",
+    static_details="Processing premium order with expedited shipping"
+)
+```
+
+The `static_summary` is a single-line description that appears in the workflow list view, while `static_details` can be multi-line and provides more comprehensive information that appears in the workflow details view.
+
+You can also use the `execute_workflow` method with the same parameters:
+
+```python
+result = await client.execute_workflow(
+    YourWorkflow.run,
+    "workflow input",
+    id="your-workflow-id",
+    task_queue="your-task-queue",
+    static_summary="Order processing for customer #12345",
+    static_details="Processing premium order with expedited shipping"
+)
+```
+
+#### Inside the Workflow
+
+Within a workflow, you can get and set the _current workflow details_. These details are distinct from the static details provided to workflow start/execute, and are included in what is returned from the `__temporal_workflow_metadata` query.
+
+```python
+@workflow.defn
+class YourWorkflow:
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Get the current details
+        current_details = workflow.get_current_details()
+        print(f"Current details: {current_details}")
+        
+        # Set/update the current details
+        workflow.set_current_details("Updated workflow details with new status")
+        
+        return "Workflow completed"
+```
+
+### Activity
+
+You can attach a summary to activities when starting them from within a workflow:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Start an activity with a summary
+        result = await workflow.execute_activity(
+            your_activity,
+            input,
+            start_to_close_timeout=timedelta(seconds=10),
+            summary="Processing user data"
+        )
+        return result
+```
+
+### Timer
+
+You can attach a summary to timers within a workflow:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Create a timer with a summary
+        await workflow.sleep(timedelta(minutes=5), summary="Waiting for payment confirmation")
+        return "Timer completed"
+```
+
+### Signal Handler
+
+When defining a signal handler in a workflow, you can include a description:
```

Comment:
So while "descriptions" were done as part of the user metadata project, I am not sure they should be considered "user metadata" and documented as such. It was just part of the project because we were getting workflow information via query anyways. If we do want to document that you can set descriptions on these handlers, we should do so where we discuss the defining of those handlers (but that can be separate from this PR).

Reply:
removed.